src/js/components/RadioButtonGroup/__tests__/__snapshots__/RadioButtonGroup-test.js.snap

```diff
@@ -408,7 +408,9 @@ exports[`RadioButtonGroup boolean options 1`] = `
           </svg>
         </div>
       </div>
-      <span>
+      <span
+        className=""
+      >
```

Comment:
Existing snap tests should not change, so the snapshot should stay intact with only code added and not code removed. One reason for it may be that the test you added wasn't being added to the end of the test file, so try that,  if that doesn't fix the snapshot, then this PR may not be backward compatible and we'll need to find the root cause for it and fix it.

Reply:
Hrmmm, from the research I've done (I'm new to styled-components) - it looks like empty classes are pretty common when moving from non-styled components to styled components (in the event the logic dictates that no css is written). Here, we have a new styled.span that only renders css if radioButton.fontWeight is specified. So the default case (radioButton.fontWeight = undefined) renders no css, and thus an empty className="" (I'm guessing, based on my new understanding of styled-components).

The reason RadioButtonGroup-test.js.snap changed is because RadioButtonGroup uses RadioButton under the hood.

I did try moving it to the end of the tests, but that doesn't fix it.

I guess I could add theme checking logic in RadioButton.js (if theme.radioButton.fontWeight { render new StyledLabel } else render old span with no empty className, but not sure if it's just overcomplicating things.

Reply:
Thanks for sharing these great insights. I think I'm OK with how it works now.